src/SSHDebugPS/ConnectionManager.cs

```diff
@@ -31,6 +31,7 @@ public static DockerConnection GetDockerConnection(string name, bool supportSSHC
             DockerContainerTransportSettings settings;
             Connection remoteConnection;
 
+            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
>ThreadHelper.ThrowIfNotOnUIThread(); [](http://example.com/codeflow?start=12&length=36)

I am not sure this will always be called on the UI thread. Maybe it is true, but it is difficult to prove because the result of this is cached in the port object.

What was the thing complaining that caused you to add it?

Reply:
```
##[warning]src\SSHDebugPS\ConnectionManager.cs(80,111): Warning VSTHRD010: Accessing "ServiceProvider" should only be done on the main thread. Call Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread() first.
C:\a\1\s\src\SSHDebugPS\ConnectionManager.cs(80,111): warning VSTHRD010: Accessing "ServiceProvider" should only be done on the main thread. Call Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread() first. [C:\a\1\s\src\SSHDebugPS\SSHDebugPS.csproj]
##[warning]src\SSHDebugPS\ConnectionManager.cs(80,96): Warning VSTHRD010: Accessing "ServiceProvider" should only be done on the main thread. Call Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread() first.
C:\a\1\s\src\SSHDebugPS\ConnectionManager.cs(80,96): warning VSTHRD010: Accessing "ServiceProvider" should only be done on the main thread. Call Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread() first. [C:\a\1\s\src\SSHDebugPS\SSHDebugPS.csproj]
```

Reply:
Ah, should I just suppress it then with a comment?

Reply:
I think your fix is better than what we had before, at least if you aren't seeing any issues when you test. It does seem better to throw here instead of potentially asking the service provider for the connection manager on a background thread.

We don't have the ideal design here, but hopefully your fix is good enough.